Source/Core/Core/PowerPC/Tables/TableGen.cpp

```diff
@@ -0,0 +1,476 @@
+          << ",\n";
+    }
+    out << "End = " << table.size() + 1 << '\n';
+  }
+  if (options.flags & DECODING_TABLE)
+  {
+    for (auto& dec : decoding_table)
+    {
+      out << "{0b" << dec.instructions << ", 0b" << dec.subtables << ", "
+          << dec.instruction_offset + 1 << ", " << dec.subtable_offset << ", " << dec.shift << ", "
+          << dec.length << "},\n";
+    }
+  }
+  if (!options.columns.empty())
+  {
+    for (auto& inst : table)
+    {
+      if (options.columns.size() > 1)
+      {
+        out << "{";
+      }
+      for (auto& col : options.columns)
+      {
+        if (inst.cells.size() <= col.column || inst.cells[col.column].empty())
+        {
+          if (col.default_value == "*")
+          {
+            out << col.prefix << inst.cells.at(0) << col.suffix << ", ";
+          }
+          else
+          {
+            out << col.default_value << ", ";
+          }
+        }
+        else if (inst.cells[col.column] == "*")
+        {
+          out << col.prefix << inst.cells[0] << col.suffix << ", ";
+        }
+        else
+        {
+          out << col.prefix << inst.cells[col.column] << col.suffix << ", ";
+        }
+      }
+      if (options.columns.size() > 1)
+      {
+        out << "},";
+      }
+      out << '\n';
+    }
+  }
+  if ((options.flags & OPID_RANGES) || (options.flags & DECODING_TABLE) || !options.columns.empty())
+  {
+    out << "// clang-format on\n";
+  }
+}
+
+static void ReadTable(std::istream& in, std::vector<std::vector<std::string>>& rows)
+{
+  while (!in.eof())
+  {
+    std::string line;
+    std::getline(in, line);
+    std::vector<std::string>::size_type nextpos, pos = 0;
+    std::vector<std::string> row;
+    if (line.empty())
+    {
+      rows.push_back(std::move(row));
+      continue;
+    }
+    while (true)
+    {
+      nextpos = line.find(SEPARATOR, pos);
+      if (nextpos != std::string::npos)
+      {
+        row.push_back(line.substr(pos, nextpos - pos));
+        pos = nextpos + 1;
+      }
+      else
+      {
+        row.push_back(line.substr(pos, std::string::npos));
+        break;
+      }
+    }
+    rows.push_back(std::move(row));
+  }
+}
+
+static void ParseCommandLine(int argc, char** argv, OutputOptions& stdout_options,
+                             std::vector<std::pair<std::ofstream, OutputOptions>>& file_outputs,
+                             std::string& inputfile)
+{
+  if (argc == 1)
+  {
+    const char* const help_string =
+        " [options]\n"
+        "Options:\n"
+        "-i <file>   read from this file, not stdin"
+        "-o <file>   sets an output file for the preceding options. Trailing options\n"
+        "            are applied for stdout.\n"
+        "-r          generate OpID range definition\n"
+        "-D          generate decoding table\n"
+        "-0…9        Generate a custom table. Add the column specified by the digit.\n"
+        "-p <prefix>  Set the prefix for the previously-defined column."
```

Comment:
Missing newlines on the -p, -i, -s. Consider indenting the options, maybe?

Reply:
You mean insert spaces in front of the dash?